commons/src/main/java/org/infinispan/commons/api/functional/Closeables.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.commons.api.functional;
+
+import java.util.Spliterator;
+
+// TODO: Remove once Will's branch closeable spliterator is in.
+
+/**
+ * @since 8.0
+ */
+public class Closeables {
+
+   private Closeables() {
+      // Cannot be instantiated, it's just a holder class
+   }
+
+   /**
+    * Auto closeable spliterator. Being able to close an spliterator is useful
+    * for situations where the iterator might consume considerable resources,
+    * e.g. in a distributed environment. Hence, being able to close it allows
+    * for resources to be reclaimed if the iterator won't be iterated any more.
+    */
+   public interface CloseableSpliterator<T> extends Spliterator<T>, AutoCloseable {
```

Comment:
@danberindei It would be just like you would normally expect
```
try (CloseableSpliterator<V> spliterator = cache.values().spliterator()) {
    // Do whatever you need with the spliterator
}
```

The only other way I can think to make this explicit and still work even when using Map interface is to prevent anyone from calling iterator or spliterator by throwing an UnsupportedOperationException and then subsequently point them to call stream.iterator like the following:

```
try (Stream<V> stream = map.values().stream()) {
   Iterator<V> iterator = stream.iterator();
}
```